crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -415,9 +416,11 @@ where
         to_block: u64,
     ) -> Result<Vec<Log>, FilterError> {
         trace!(target: "rpc::eth::filter", from=from_block, to=to_block, ?filter, "finding logs in range");
+        let info = self.provider.chain_info()?;
+        let best_number = info.best_number;
```

Comment:
hmm, I believe we can pass `ChainInfo` as an additional argument to the function because we should already have this when we call this function.